python/pyspark/tests.py

```diff
@@ -1859,6 +1860,31 @@ def test_with_different_versions_of_python(self):
         finally:
             self.sc.pythonVer = version
 
+    def test_exception_blocking(self):
+        """
+        SPARK-21045
+        Test whether program is blocked when occur exception in worker sending
+        exception to PythonRDD
+
+        """
+        import threading
+
+        def run():
+            try:
+
+                def f():
+                    raise Exception("中")
+
+                self.sc.parallelize([1]).map(lambda x: f()).count()
+            except Exception:
+                pass
+
+        t = threading.Thread(target=run)
```

Comment:
This test mainly check whether it is blocked. So I use thread join, if it is blocked before fixing the bug the program will wait 10s and exit instead blocking other tests.